panic: runtime error: invalid memory address or nil pointer dereference

**How to categorize this issue?**
<!--
Please select area, kind, and priority for this issue. This helps the community categorizing it.
Replace below TODOs or exchange the existing identifiers with those that fit best in your opinion.
If multiple identifiers make sense you can also state the commands multiple times, e.g.
  /area control-plane
  /area auto-scaling
  ...

"/area" identifiers:     audit-logging|auto-scaling|backup|certification|control-plane-migration|control-plane|cost|delivery|dev-productivity|disaster-recovery|documentation|high-availability|logging|metering|monitoring|networking|open-source|operations|ops-productivity|os|performance|quality|robustness|scalability|security|storage|testing|usability|user-management
"/kind" identifiers:     api-change|bug|cleanup|discussion|enhancement|epic|impediment|poc|post-mortem|question|regression|task|technical-debt|test
"/priority" identifiers: 1|2|3|4|5 (ordered from greatest to least)
-->
/area control-plane-migration
/kind bug
/priority 3

**What happened**:

gardenlet is `CrashLoopBackOff` because of a panic in the controlplane migration flow.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x1d18769]

goroutine 4641 [running]:
github.com/gardener/gardener/pkg/operation/botanist.(*Botanist).setAPIServerServiceClusterIP(0xc003ac3ec0, 0xc00ee1c230, 0xc)
	/go/src/github.com/gardener/gardener/pkg/operation/botanist/controlplane.go:987 +0x289
github.com/gardener/gardener/pkg/operation/botanist/controlplane.(*kubeAPIService).Deploy(0xc00f02a0b0, 0x25bb340, 0xc000056018, 0xc00f02a0b0, 0x0)
	/go/src/github.com/gardener/gardener/pkg/operation/botanist/controlplane/kube_apiserver_service.go:138 +0x255
github.com/gardener/gardener/pkg/operation/botanist.(*Botanist).DeployKubeAPIService(0xc003ac3ec0, 0x25bb340, 0xc000056018, 0x1, 0x0, 0x0)
	/go/src/github.com/gardener/gardener/pkg/operation/botanist/controlplane.go:953 +0x5d
github.com/gardener/gardener/pkg/operation/botanist.(*Botanist).WakeUpKubeAPIServer(0xc003ac3ec0, 0x25bb340, 0xc000056018, 0xa46c8bc, 0xc002ee7da0)
	/go/src/github.com/gardener/gardener/pkg/operation/botanist/controlplane.go:213 +0x85
github.com/gardener/gardener/pkg/utils/flow.(*execution).runNode.func1(0xc0077749a0, 0x2326781, 0x38, 0x25bb340, 0xc000056018)
	/go/src/github.com/gardener/gardener/pkg/utils/flow/flow.go:219 +0x1d5
created by github.com/gardener/gardener/pkg/utils/flow.(*execution).runNode
	/go/src/github.com/gardener/gardener/pkg/utils/flow/flow.go:214 +0x14f
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- Gardener version:

```
version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.1", GitCommit:"fb648412afde6ff8a1e8d1f92b4378f3c88d8d35", GitTreeState:"clean", BuildDate:"2021-03-22T14:49:03+00:00", GoVersion:"go1.15.9", Compiler:"gc", Platform:"linux/amd64"}
```

- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

--- a/pkg/gardenlet/controller/shoot/shoot_control_migrate.go
+++ b/pkg/gardenlet/controller/shoot/shoot_control_migrate.go
@@ -82,7 +82,6 @@ func (c *Controller) prepareShootForMigration(ctx context.Context, logger *logru
 func (c *Controller) runPrepareShootControlPlaneMigration(o *operation.Operation) *gardencorev1beta1helper.WrappedLastErrors {
 	var (
 		ctx                          = context.TODO()
-		namespace                    = &corev1.Namespace{}
 		botanist                     *botanistpkg.Botanist
 		err                          error
 		tasksWithErrors              []string
@@ -114,7 +113,7 @@ func (c *Controller) runPrepareShootControlPlaneMigration(o *operation.Operation
 		}),
 		utilerrors.ToExecute("Retrieve kube-apiserver deployment in the shoot namespace in the seed cluster", func() error {
 			deploymentKubeAPIServer := &appsv1.Deployment{}
-			if err := botanist.K8sSeedClient.Client().Get(ctx, kutil.Key(o.Shoot.SeedNamespace, v1beta1constants.DeploymentNameKubeAPIServer), deploymentKubeAPIServer); err != nil {
+			if err := botanist.K8sSeedClient.APIReader().Get(ctx, kutil.Key(o.Shoot.SeedNamespace, v1beta1constants.DeploymentNameKubeAPIServer), deploymentKubeAPIServer); err != nil {
 				if !apierrors.IsNotFound(err) {
 					return err
 				}
@@ -126,13 +125,15 @@ func (c *Controller) runPrepareShootControlPlaneMigration(o *operation.Operation
 			return nil
 		}),
 		utilerrors.ToExecute("Retrieve the Shoot namespace in the Seed cluster", func() error {
-			if err := botanist.K8sSeedClient.Client().Get(ctx, client.ObjectKey{Name: o.Shoot.SeedNamespace}, namespace); err != nil {
+			botanist.SeedNamespaceObject = &corev1.Namespace{}
+			err := botanist.K8sSeedClient.APIReader().Get(ctx, client.ObjectKey{Name: o.Shoot.SeedNamespace}, botanist.SeedNamespaceObject)
+			if err != nil {
 				if apierrors.IsNotFound(err) {
 					o.Logger.Infof("Did not find '%s' namespace in the Seed cluster - nothing to be done", o.Shoot.SeedNamespace)
 					return utilerrors.Cancel()
 				}
 			}
-			return nil
+			return err
 		}),
 	)
 
@@ -144,7 +145,7 @@ func (c *Controller) runPrepareShootControlPlaneMigration(o *operation.Operation
 	}
 
 	var (
-		nonTerminatingNamespace = namespace.Status.Phase != corev1.NamespaceTerminating
+		nonTerminatingNamespace = botanist.SeedNamespaceObject.Status.Phase != corev1.NamespaceTerminating
 		cleanupShootResources   = nonTerminatingNamespace && kubeAPIServerDeploymentFound
 		wakeupRequired          = (o.Shoot.Info.Status.IsHibernated || (!o.Shoot.Info.Status.IsHibernated && o.Shoot.HibernationEnabled)) && cleanupShootResources
 		defaultTimeout          = 10 * time.Minute

